newrelic/resource_newrelic_alert_muting_rule.go

```diff
@@ -1,13 +1,77 @@
 package newrelic
 
 import (
-	"log"
-
+	"fmt"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 	"github.com/newrelic/newrelic-client-go/pkg/errors"
+	"log"
+	"time"
 )
 
+func validateNaiveDateTime(val interface{}, key string) (warns []string, errs []error) {
+	valueString := val.(string)
+
+	// test conversion to desired format:
+	_, err := time.Parse("2006-01-02T15:04:05", valueString)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("%#v of %#v must be in the format 2006-01-02T15:04:05", key, valueString))
+	}
+	return
+}
+
+func scheduleSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"end_repeat": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The datetime stamp when the MutingRule schedule should stop repeating.",
+				ConflictsWith: []string{"schedule.repeat_count"},
+				ValidateFunc:  validateNaiveDateTime,
+			},
+			"end_time": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The datetime stamp representing when the MutingRule should end.",
+				ValidateFunc: validateNaiveDateTime,
+			},
+			"repeat": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The frequency the MutingRule schedule repeats. One of [DAILY, WEEKLY, MONTHLY]",
+				ValidateFunc: validation.StringInSlice([]string{"DAILY", "WEEKLY", "MONTHLY"}, false),
+			},
```

Comment:
What is the `false` argument doing?  I'm not sure even looking at the function definition.

Reply:
Ah, yeah, false is for the 'ignoreCase' parameter to this function.  At first I thought "yeah, we want to ignore case to be more flexible, since nerdgraph doesn't care," but then I realized that nerdgraph is going to be sending back ALLUPPERCASE values, which is going to cause state drift if they user has things configured that don't match.

There might be something we could do to tell it to ignore case when diffing, but I just went with the simpler route of requiring screaming snake case

Reply:
I think that's the right call.  Thanks for explaining!